### PR TITLE
Fixed issue #17740: Script : Set for all languages didn't work

### PR DESCRIPTION
**Pull request by gabrieljenik**
Depending on when is merged, we need to review the version and the merge conflicts.
Added a new field at question level "same_script"
Included that at import level, export level, screen level and rendering questions level.

### CODE REVIEW

**Comment by Shnoulle** [comment]
Then : finally : it's a new feature not an issue ?

**Review by Shnoulle — APPROVED** [review]
:+1: except the DB update conflict

**Comment by c-schmitz** [comment]
This patch only be merged after release of 5.3 and then it will need an update on the database update routines.

**Review by c-schmitz — CHANGES_REQUESTED** [review]
Update script needs  to be fixed

**Comment by gabrieljenik** [comment]
Ready for re-reviewal

File: application/models/QuestionBaseRenderer.php [hunk 1]
```diff
@@ -59,8 +59,9 @@ public function __construct($aFieldArray, $bRenderDirect = false)
         $oQuestionTemplate = QuestionTemplate::getNewInstance($this->oQuestion);
         $oQuestionTemplate->registerAssets(); // Register the custom assets of the question template, if needed
 
-        if (!empty($this->oQuestion->questionl10ns[$this->sLanguage]->script)) {
-            $sScriptRendered = LimeExpressionManager::ProcessString($this->oQuestion->questionl10ns[$this->sLanguage]->script, $this->oQuestion->qid, ['QID' => $this->oQuestion->qid]);
+        $questionScript = $this->getQuestionScript();
+        if (!empty($questionScript)) {
+            $sScriptRendered = LimeExpressionManager::ProcessString($questionScript, $this->oQuestion->qid, ['QID' => $this->oQuestion->qid]);
```
Comment (Shnoulle) [inline application/models/QuestionBaseRenderer.php:64]: We get dynamic script here ? Must use only static expression, no ? (another issue)

File: application/views/questionAdministration/textElements.php [hunk 2]
```diff
@@ -94,19 +94,26 @@
...
+                                    <?php endif; ?>
+                                />&nbsp;
+                                <label for="same_script">
+                                    <?= gT('Use for all languages'); ?>
+                                </label>
+                            </div>
                         </div>
-                    </div> 
+                    <?php else: ?>
+                        <div class="alert alert-warning same-script-alert" style="display: none;"><?= gT('The script for this language will not be used because "Use for all languages" is set on the base language\'s script.') ?></div>
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/views/questionAdministration/textElements.php:115]: use of hidden class ? Then admin theme can hide/show on the different way using css animate ?

File: assets/scripts/admin/questionEditor.js [hunk 3]
```diff
@@ -1934,6 +1934,14 @@ $(document).on('ready pjax:scriptcomplete', function () {
     $('#general-setting-help-relevance').show();
   }
 
+  function showSameScriptForAllLanguagesWarning() {
+    if ($('#same_script').is(":checked")) {
+      $('.same-script-alert').show();
+    } else {
+      $('.same-script-alert').hide();
```
Comment (Shnoulle) [inline assets/scripts/admin/questionEditor.js:1941]: .addClass("hidden")

**Review by Shnoulle — APPROVED** [review]
For perfection : usage of hidden

<s>Else : need to report the issue about use tag inside script ?</s>

File: application/models/QuestionBaseRenderer.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/QuestionBaseRenderer.php:64]: @tpartner see issue https://bugs.limesurvey.org/view.php?id=18024 about Script in static or dynamic :)

**Comment by gabrieljenik** [comment]
> For perfection : usage of hidden

Perfect now! :)

**Comment by Shnoulle** [comment]
> > For perfection : usage of hidden
> 
> Perfect now! :)

Already approved ;)

**Comment by glimz** [comment]
@gabrieljenik you can fix the conflicts and merge.

**Comment by gabrieljenik** [comment]
@c-schmitz Is it ok to merge? @glimz says I can merge, but your review is requested and pending. Thanks!

File: installer/create-database.php [hunk 4]
```diff
@@ -381,7 +381,8 @@ function populateDatabase($oDB)
             'same_default' =>  "integer NOT NULL default '0'",
             'relevance' =>  "text",
             'question_theme_name' => "string(150) NULL",
-            'modulename' =>  "string(255) NULL"
+            'modulename' =>  "string(255) NULL",
+            'same_script' => "integer NOT NULL default '0'",
```
Comment (c-schmitz) [inline installer/create-database.php:385]: same here

**Review by c-schmitz — CHANGES_REQUESTED** [review]

**Comment by c-schmitz** [comment]
If resolved, please merge.

File: installer/create-database.php (hunk 4 above)
Comment (gabrieljenik) [reply installer/create-database.php:385]: Look at line 381 (3 lines above). It is written the same way. I believe it works either way.

File: application/helpers/update/updates/Update_487.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+{
+    /**
+     * @inheritDoc
+     */
+    public function up()
+    {
+        $this->db->createCommand()->addColumn(
+            '{{questions}}',
+            'same_script',
+            "integer NOT NULL default '0'"
```
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/helpers/update/updates/Update_487.php:15]: I believe it works either way. It is written like that all over the application.

**Comment by Shnoulle** [comment]
@gabrieljenik can you fix the confict ? I merge it myself after.

**Comment by Shnoulle** [comment]
Not related to this issue, but the animate on alert is really awfull  …
Peek 15-06-2022 09-03

**Comment by Shnoulle** [comment]
Tested : 

1. Edit
2. import LSS from old version
3. import LSQ from old verison
4. export/import from current (LSS)
5. export/import from current (LSQ)

All seems OK, i merge.

I have a new bug to submit about some HTML improvement :)

**Pull request merged**